ckanext/s3filestore/uploader.py

```diff
@@ -183,24 +211,40 @@ def update_data_dict(self, data_dict, url_field, file_field, clear_field):
         self.clear = data_dict.pop(clear_field, None)
         self.file_field = file_field
         self.upload_field_storage = data_dict.pop(file_field, None)
+        self.upload_file = None
+        self.preserve_filename = data_dict.get('preserve_filename', None)
```

Comment:
Should we make preserve_filename an extra param to allow it to be changed on load following the others 
i.e
update_data_dict(self, data_dict, url_field, file_field, clear_field, preserve_filename_field='preserve_filename')
and 
self.preservce_filename = data_dict.get('preserve_filename_field, None(

Reply:
I'm not sure; I didn't write that part. What does the flag determine?